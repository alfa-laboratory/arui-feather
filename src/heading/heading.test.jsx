/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { shallow } from 'enzyme';

import Heading from './heading';

describe('heading', () => {
    it('should render without problem', () => {
        let heading = shallow(<Heading size='s'>Heading</Heading>);

        expect(heading).toMatchSnapshot();
    });

    let sizes = ['s', 'm', 'l', 'xl'];
    let headingLevel = {
        xl: 1,
        l: 2,
        m: 3,
        s: 4
    };

    sizes.forEach(size => (
        it(`should render Heading size=\`${size}\` with tag H${headingLevel[size]}`, () => {
            let heading = shallow(<Heading size={ size }>Heading</Heading>);

            expect(heading.is(`h${headingLevel[size]}`));
        })
    ));
});
