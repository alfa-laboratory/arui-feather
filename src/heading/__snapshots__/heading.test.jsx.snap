// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heading should render without problem 1`] = `
<h4
  className="heading heading_size_s heading_theme_alfa-on-white"
>
  Heading
</h4>
`;
